core/block.go

```diff
@@ -39,6 +39,24 @@ type Header struct {
 	Signatures [][]*felt.Felt
 	// Amount of STRK charged per Gas spent
 	GasPriceSTRK *felt.Felt
+	// The mode of the L1 data availability
+	L1DAMode L1DAMode
+	// The gas price for L1 data availability
+	L1DataGasPrice *GasPrice
+	// The gas price for L1
+	L1GasPrice *GasPrice
+}
+
+type L1DAMode string
+
+const (
+	CallData L1DAMode = "CALLDATA"
+	Blob     L1DAMode = "BLOB"
+)
```

Comment:
our `core` enums are all integers if I am not mistaken, can we make this something similar to;

```golang
type Resource uint32

const (
	ResourceL1Gas Resource = iota + 1
	ResourceL2Gas
)

func (r Resource) String() string {
	switch r {
	case ResourceL1Gas:
		return "L1_GAS"
	case ResourceL2Gas:
		return "L2_GAS"
	default:
		return ""
	}
}
```